File: app/src/main/java/com/ysr/lucky/utils/TUtil.java
package com.ysr.lucky.utils;

import java.lang.reflect.ParameterizedType;

/**
 * 类转换初始化
 * @author ysr
 * @data 2017/10/21 下午1:57.
 * 邮箱 yfeng1023@gmail.com
 */

public class TUtil {
    public static <T> T getT(Object o, int i) {
        try {
            return ((Class<T>) ((ParameterizedType) (o.getClass()
                    .getGenericSuperclass())).getActualTypeArguments()[i])
                    .newInstance();
        } catch (InstantiationException e) {
        } catch (IllegalAccessException e) {
        } catch (ClassCastException e) {
        }
        return null;
    }

    public static Class<?> forName(String className) {
        try {
            return Class.forName(className);
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: app/src/main/java/com/ysr/lucky/utils/TUtil.kt
package com.ysr.lucky.utils

import java.lang.reflect.ParameterizedType

/**
 * 类转换初始化
 * @author ysr
 * @data 2017/10/21 下午1:57.
 * 邮箱 yfeng1023@gmail.com
 */

object TUtil {
    fun <T> getT(o: Any, i: Int): T? {
        try {
            return ((o.javaClass
                    .genericSuperclass as ParameterizedType).actualTypeArguments[i] as Class<T>)
                    .newInstance()
        } catch (e: InstantiationException) {
        } catch (e: IllegalAccessException) {
        } catch (e: ClassCastException) {
        }

        return null
    }

    fun forName(className: String): Class<*>? {
        try {
            return Class.forName(className)
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        }

        return null
    }
}
